### PR TITLE
(fix) 03-2455: Show delete confirmation modal upon deleting a patient list

### PR DESCRIPTION
**Pull request by jwnasambu**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
I added a show delete confirmation modal upon deleting a patient list. To achieve this, I added the showDeleteConfirmationModal state variable to control the visibility of the delete confirmation modal. The handleDelete function sets this state to true, and the modal is displayed when showDeleteConfirmationModal is true. The confirmDeletePatientList function handles the actual deletion and closes the modal when done.

## Screenshots
**Before Making changes**

https://github.com/openmrs/openmrs-esm-patient-management/assets/33891016/c8b1fdba-a00e-4933-895c-31e149ca9e7b

**After making changes**

https://github.com/openmrs/openmrs-esm-patient-management/assets/33891016/177e7d7b-0fec-4c41-bd81-96a4c81a536b

## Related Issues
https://issues.openmrs.org/browse/O3-2455

## Other

*None.*

### CODE REVIEW

File: packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.component.tsx [hunk 1]
```diff
@@ -122,10 +128,16 @@ const PatientListDetailComponent = () => {
...
             <OverflowMenuItem
-              itemText={t('editNameDescription', 'Edit Name/ Description')}
+              className={styles.menuItem}
+              itemText={t('editNameDescription', 'Edit Name or Description')}
               onClick={() => setEditPatientListDetailOverlay(true)}
             />
-            <OverflowMenuItem itemText={t('delete', 'Delete')} onClick={handleDelete} isDelete />
+            <OverflowMenuItem
+              className={styles.menuItem}
+              itemText={t('deletePatientList', 'Delete patient list')}
```
Comment (ibacher) [inline packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.component.tsx:137]: I don't see this string in the translation files. Make sure you run the git commit hooks.

File: packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.test.tsx [hunk 2]
```diff
@@ -82,19 +82,31 @@ describe('PatientListDetailComponent', () => {
...
     render(<PatientListDetailComponent />);
 
     await waitFor(() => {
-      userEvent.click(screen.getByText('Delete'));
+      userEvent.click(screen.getByText('Actions'));
+      userEvent.click(screen.getByText('Delete patient List'));
+    });
+
+    await waitFor(() => {
+      expect(screen.getByText('Are you sure you want to delete the patient list')).toBeInTheDocument();
```
Comment (ibacher) [inline packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.test.tsx:102]: Note that this string will not match `Are you sure you want to delete {patientListName} patient list?`.

File: packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.component.tsx (hunk 1 above)
Comment (hadijahkyampeire) [reply packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.component.tsx:137]: @jwnasambu fixing Ian's comment above will help fix the failing test.

File: packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.test.tsx (hunk 2 above)
Comment (denniskigen) [reply packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.test.tsx:102]: Just a thought - a possible variation of this translation that’s more flexible is: Are you sure you want to delete the list named {listName}

Comment (jwnasambu) [reply packages/esm-patient-list-app/src/patient-list-detail/patient-list-detail.test.tsx]: @ibacher, @denniskigen and @hadijahkyampeire thanks for the review and guidance.

**Review by denniskigen — APPROVED** [review]

**Comment by denniskigen** [comment]
Thanks, @jwnasambu!

**Comment by denniskigen** [comment]
A nice next step would be updating the E2E test.

**Pull request merged**